src/auto-complete.jsx

```diff
@@ -0,0 +1,266 @@
+        listStyle={this.mergeAndPrefix(styles.list, listStyle)}
+        openDirection="bottom-left"
+        style={mergedMenuStyles}>
+        {
+          requestsList.map((request, index) => {
+            switch(typeof request){
+              case 'string':
+                return (<MenuItem
+                          disableFocusRipple={this.props.disableFocusRipple}
+                          innerDivStyle={{overflow:'hidden'}}
+                          key={index}
+                          value={request}
+                          primaryText={request}
+                          />);
+              case 'object':
+                return React.cloneElement(request, {
+                        key: index,
+                        disableFocusRipple: this.props.disableFocusRipple,
+                      });
+              default:
+                return null;
+            }
+          })
+        }
+      </Menu>
+    ) : null;
+
+    return (
+      <div style={mergedRootStyles}
+           onKeyDown={(e)=>{
+            switch(e.keyCode){
+              case KeyCode.ESC:
+                this.setState({open:false});
+                break;
+              case KeyCode.DOWN:
+                if(this.focusOnInput && this.state.open){
+                  e.preventDefault();
+                  this.focusOnInput = false;
+                  this.setState({open:true});
+                }
+                break;
+              default:
+                break;
+            }
+          }}>
+        <div
+          style={{
+            widht:'100%',
+          }}>
+          <TextField
+            {...other}
+            ref="searchTextField"
+            value={this.state.searchText}
+            onEnterKeyDown={()=>{
+                 setTimeout(() => {
+                   this.state({open:false});
+                 }, this.props.touchTapCloseDelay);
+                this.props.onNewRequest(this.state.searchText);
+            }}
+            onChange={(e)=>{
+                let searchText = e.target.value;
+                this.updateRequests(searchText);
+            }}
+            onBlur={()=>{
+              if(this.focusOnInput && this.state.open)
+                this.refs.searchTextField.focus();
+            }}
+            onFocus={()=>{
+              if(!this.state.open && (
+                  this.props.updateWhenFocused || this.searchText !== '')){
+                this.updateRequests(this.state.searchText);
+              }
+              this.focusOnInput = true;
+            }}
+
+            {...textFieldProps} />
+        </div>
+          <ReactTransitionGroup>{menu}</ReactTransitionGroup>
+      </div>
+    );
+  },
+
+  setValue(textValue) {
+    this.setState({
+      searchText: textValue,
+    });
+  },
+
+  getValue() {
+    return this.state.searchText;
+  },
+
+  updateRequests(searchText){
+    let requestsListPromise = new Promise((resolve, reject) => {
+      let list = this.props.onUpdateRequests(searchText);
```

Comment:
Additionally, `onUpdateRequests` does not actually support returning a `Promise`. The `Promise` is internal to `updateRequests` which relies on `onUpdateRequests` to return the list of items to render. Returning a `Promise` from `onUpdateRequests` would throw an error because its expecting an array with which to `map` over. `Promise`s obviously don't have a `map` method.

If you expect `onUpdateRequests` to return a `Promise`, then this should probably be re-written as:
```
let requestsListPromise = this.props.onUpdateRequests(searchText);
```

But I agree with @oliviertassinari, the component should accept a `list` or `items` property. It'd be more idiomatic React that way.